provisioner/provision.js

```diff
@@ -198,6 +202,37 @@ Provisioner.prototype.runAllProvisionersOnce = function () {
 };
 
 /**
+ * Figure out how to create the launch information based on a bid then
+ * insert the secrets into the secret storage
+ */
+Provisioner.prototype.spawn = function (workerType, bid) {
+  assert(workerType);
+  assert(bid);
+
+  var launchInfo = workerType.createLaunchSpec(bid.region, bid.type);
+
+  // This should probably move from here to createLaunchSpec but that
+  // can happen later
+  launchInfo.launchSpec.Placement = {
+    AvailabilityZone: bid.zone,
+  };
+
+  // TODO: Generate temporary credentials and stick 'em in around about
+  //       here
```

Comment:
No, do this when people get the secrets... No need to store the credentials in azure table storage when we can settle with just storing the scopes...